test/utxo-select.test.js

```diff
@@ -967,6 +970,58 @@ describe('UTXO Select', () => {
       fee: 2591n,
       change: false,
     });
+
+    const t3 = (strategy) => {
+      const FEE = 0n // no fee to test exact amounts
+      const est = new btc._Estimator(inputs, [{ ...OUTPUTS[0], amount: inputsTotalAmount }], {
+        feePerByte: FEE,
+        changeAddress: '2MshuFeVGhXVdRv77UcJYvRBi2JyTNwgSR2',
+        network: regtest,
+        allowLegacyWitnessUtxo: true,
+        createTx: true,
+        allowSameUtxo: true,
+      });
+      const acc = est.result(strategy);
+      if (!acc) return;
+      const i = acc.inputs.map((i) => i.witnessUtxo.amount);
+      const o = acc.outputs.map((i) => i.amount);
+      const tx = acc.tx;
+      tx.sign(privKey1);
+      tx.finalize();
+      const expFee = BigInt(tx.vsize) * FEE;
+      return {
+        i,
+        o,
+        txFee: tx.fee,
+        expFee,
+        fee: acc.fee,
+        change: acc.change,
+      };
+    };
+    deepStrictEqual(t3('default'), {
+      i: inputs.map((i) => i.witnessUtxo.amount).reverse(),
+      o: [inputsTotalAmount],
+      txFee: 0n,
+      expFee: 0n,
+      fee: 0n,
+      change: false,
+    });
+    deepStrictEqual(t3('accumSmallest'), {
+      i: inputs.map((i) => i.witnessUtxo.amount),
+      o: [inputsTotalAmount],
+      txFee: 0n,
+      expFee: 0n,
+      fee: 0n,
+      change: false,
+    });
+    deepStrictEqual(t3('all'), {
+      i: inputs.map((i) => i.witnessUtxo.amount),
+      o: [inputsTotalAmount],
+      txFee: 0n,
+      expFee: 0n,
+      fee: 0n,
+      change: false,
+    });
```

Comment:
New tests to verify that `input amounts === output amounts` with 0 fees results in all inputs selected. This is logically sound with regard to how accumulate should work imo.